addons/google_spreadsheet/models/google_drive_config.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import cgi
+import json
+import logging
+from lxml import etree
+import urllib2
+import werkzeug
+
+from odoo import api, models, _
+from odoo.addons.google_account import TIMEOUT
+
+_logger = logging.getLogger(__name__)
+
+
+class Config(models.Model):
+    _inherit = 'google.drive.config'
+
+    def get_google_scope(self):
+        scope = super(Config, self).get_google_scope()
+        return '%s https://spreadsheets.google.com/feeds' % scope
+
+    @api.model
+    def write_config_formula(self, attachment_id, spreadsheet_key, model, domain, groupbys, view_id):
+        access_token = self.get_access_token(scope='https://spreadsheets.google.com/feeds')
+
+        fields = self.env[model].fields_view_get(view_id=view_id, view_type='tree')
+        doc = etree.XML(fields.get('arch'))
+        display_fields = []
+        for node in doc.xpath("//field"):
+            if node.get('modifiers'):
+                modifiers = json.loads(node.get('modifiers'))
+                if not modifiers.get('invisible') and not modifiers.get('tree_invisible'):
+                    display_fields.append(node.get('name'))
+        fields = " ".join(display_fields)
+
+        domain = domain.replace("'", r"\'").replace('"', "'")
+        if groupbys:
+            fields = "%s %s" % (groupbys, fields)
+            formula = '=oe_read_group("%s";"%s";"%s";"%s")' % (model, fields, groupbys, domain)
+        else:
+            formula = '=oe_browse("%s";"%s";"%s")' % (model, fields, domain)
+
+        url = self.env['ir.config_parameter'].get_param('web.base.url')
+        if not self.env.user.password:
+            config_formula = '=oe_settings("%s";"%s")' % (url, self.env.cr.dbname)
+        else:
+            config_formula = '=oe_settings("%s";"%s";"%s";"%s")' % (url, self.env.cr.dbname, self.env.user.login, self.env.user.password)
+
+        template = self.env.ref('google_spreadsheet.sheets_api_request_format')
+        feed = etree.fromstring(template.arch_base).findall('{http://www.w3.org/2005/Atom}feed')[0]
```

Comment:
plz make separate commit to refector this method to easy understand why need it.